### DISCUSSION TITLE
I want to upload a file into a gist -- not paste it in

### DISCUSSION THREAD

**Discussion by mcf-rocks** [community/community#31767] [Other Feature Feedback, Questions, & Ideas]
I want to upload a file into a gist -- not paste it in -- how?

**Comment by spenserblack** [comment]
If you're OK with using the command-line, with the [GitHub CLI](https://github.com/cli/cli) you can use `gh gist create <filename>...` to upload 1 or more files as a gist.

**Reply by mcf-rocks** [reply]
fix the website pls

**Reply by mcf-rocks** [reply]
i already have a low opinion of github... fix the website as i do not want to learn the CLI... thx.

**Reply by ThioJoe** [reply]
For those from Google, you can also update an existing Gist file using:
`gh gist edit <gist id> <file to replace with>`

It seems to retain the file name that is on the website, meaning if the file you're uploading is named differently, it won't rename the gist file.

**Comment by DenverCoder1** [comment]
GitHub has the option to drag and drop files into the editor to auto-populate the gist with those files.

![image](https://user-images.githubusercontent.com/20955511/188244278-3c593e50-f93f-44c0-8eee-6f90675df529.png)

All you have to do is drag a file from your computer into the window and the message to drop the files to prefill the gist will appear. While that message is displayed, you can drop the file and it will add it to the gist.

There are also editor extensions such as GistPad for VSCode among others that let you create gists from files without even visiting the GitHub website or using a command line interface.

**Reply by Twisted-Code** [reply]
I'm having issues doing this. When I drag it only sets the file name, and I also tried using the file picker, but for some reason it tells me my file type is not supported (even though it's .txt and that's one of the listed types). I ended up using my clipboard.

**Reply by Vojtak42** [reply]
And how can you do it on mobile?

**Reply by karl-police** [reply]
But it doesn't work.

**Comment by roke-julian-lockhart** [comment]
> https://github.com/orgs/community/discussions/31767#discussion-4356822

@mcf-rocks, I can't tell what you're asking. Do you want to be able to pre-fill a gist with a file's content, or upload that file as an attachment?

I ask because, like https://github.com/orgs/community/discussions/31767#discussioncomment-6950772, I'm unable to perform the former on a mobile device.
